stage1/init/container.go

```diff
@@ -79,12 +80,30 @@ func LoadContainer(root string) (*Container, error) {
 	return c, nil
 }
 
+// quoteExec returns an array of quoted strings appropriate for systemd execStart usage
+func quoteExec(exec []string) string {
+	var qexec []string
+
+	if len(exec) > 0 {
+		//FIXME(vc): systemd gets angry if qexec[0] is quoted, which must be a bug.
+		qexec = append(qexec, exec[0])
+
+		if len(exec) > 1 {
+			for _, arg := range exec[1:] {
+				qexec = append(qexec, strconv.Quote(arg))
+			}
+		}
+	}
+
+	return strings.Join(qexec, " ")
+}
+
 // appToSystemd transforms the provided app manifest into systemd units
 func (c *Container) appToSystemd(am *schema.ImageManifest, id types.Hash) error {
 	name := am.Name.String()
 	app := am.App
 	execWrap := []string{"/diagexec", rktpath.RelAppRootfsPath(id)}
-	execStart := strings.Join(append(execWrap, app.Exec...), " ")
```

Comment:
Right, I see. It would be nice if it did though, presonaly, I'd consider moving whatever is being implement here into the library eventually.

It would be neat if it did:
```
unit.ServiceExecStart(arrayAsIfThisWasUnixExecCall)
```
And the library would handle the multi-level shell-alike quoting/escaping hell.